_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,699 @@
+---
+Keycloak is an open-source identity and access management solution that allows you to secure your applications and services with ease. JHipster, on the other hand, is a powerful development platform that provides developers with the tools they need to create modern, scalable, and robust web applications using Spring Boot.
+
+Keycloak is the default OpenID Connect server configured with JHipster. Using Keycloak with JHipster is an excellent way to ensure that your web applications are secure and well-protected from unauthorized access. By integrating Keycloak to your applications, you can easily implement authentication and authorization, manage user identities, and provide secure access to resources.
+
+The JHipster Kubernetes generator creates all the necessary Kubernetes resources, such as deployments, services, and ingresses, based on the configuration of the JHipster application. This includes setting up the database, configuring security, and setting up any necessary environment variables. Since JHipster 8, the Kubernetes sub-generator supports Keycloak for Ingress GKE deployment. In this post, I'll walk you through the generation of a demo Spring Boot microservices application with Keycloak integration using JHipster, and its deployment to Google Kubernetes Engine (GKE).
+
+{% img blog/keycloak-kubernetes-prod/keycloak-gke.png alt:"Keycloak, JHipster, GKE logos" width:"500" %}{: .center-image }
+
+> **This tutorial was created with the following tools and services**:
+> - [Node.js v18.16.1](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+> - [npm 9.5.1](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+> - [Java OpenJDK 17](https://jdk.java.net/java-se-ri/17)
+> - [DockerHub account](https://hub.docker.com/)
+> - [Docker 24.0.2](https://docs.docker.com/desktop/)
+> - [Google Cloud CLI 437.0.1](https://cloud.google.com/sdk/docs/install)
+> - [Kubectl v1.26.3](https://kubernetes.io/docs/tasks/tools/)
+> - [Auth0 account](https://auth0.com/signup)
+> - [Auth0 CLI 1.0.0](https://github.com/auth0/auth0-cli#installation)
+
+
+{% include toc.md %}
+
+## Deploy Spring Boot microservices and Keycloak to Google Kubernetes Engine
+
+In this walkthrough, you will build a microservices architecture example from JHipster, consisting of a `gateway` application based on Spring Gateway and two Spring Boot microservices `blog` and `store`. The `gateway` will act as the entrance to your microservices, providing HTTP routing and load balancing, quality of service, security and API documentation for all microservices. With the help of the JHipster Kubernetes sub-generator, you can deploy the application and its required services (Consul for service discovery, and Keycloak for OpenID Connect) to Google Kubernetes Engine (GKE).
+
+### Build the Spring Boot microservices architecture
+
+Install JHipster, you can do the classical local installation with npm.
+
+```bash
+npm install -g generator-jhipster@8.0.0-beta.1
```

Comment:
```suggestion
npm install -g generator-jhipster@8.0.0-beta.2
```

Reply:
@mraible @deepu105  I finished working on the suggestions. I will test beta 2 now.

Reply:
@mraible @deepu105 #22718 is itll open. Will test with the development setup.